index.js

```diff
@@ -32,17 +32,16 @@ async function viaContentsApi(identifier, dir, token) {
 // Great for downloads with many sub directories
 // Pros: one request + maybe doesn't require token
 // Cons: huge on huge repos + may be truncated
-async function viaTreesApi(identifier, dir, token) {
-	if (!dir.endsWith('/')) {
-		dir += '/';
+async function viaTreesApi({user, repository, ref = 'HEAD', directory, token, getDetails = false}) {
```

Comment:
I'd rather see them on multiple lines, on both functions
```suggestion
async function viaTreesApi({
	user,
	repository,
	ref = 'HEAD',
	directory,
	token,
	getDetails = false
}) {
```